packages/embedded-provider-core/src/embedded-provider.ts

```diff
@@ -294,9 +291,15 @@ export class EmbeddedProvider {
     let session = await this.storage.getSession();
     session = await this.validateAndCleanSession(session);
 
+    // If there is no session we return null
```

Comment:
This comment literally explains what the code does and it's visible in plain sight